src/zh/baozimhorg/src/eu/kanade/tachiyomi/extension/zh/baozimhorg/BaozimhOrg.kt

```diff
@@ -49,30 +51,18 @@ class BaozimhOrg : HttpSource(), ConfigurableSource {
         .addInterceptor(UrlInterceptor)
         .build()
 
-    private fun getKey(link: String): String {
-        val pathSegments = baseHttpUrl.resolve(link)!!.pathSegments
-        val fromIndex = if (pathSegments[0] == "manga") 1 else 0
-        val toIndex = if (pathSegments.last().isEmpty()) pathSegments.size - 1 else pathSegments.size
-        val list = pathSegments.subList(fromIndex, toIndex).toMutableList()
-        list[0] = list[0].split("-").take(2).joinToString("-")
-        return list.joinToString("/")
-    }
-
     override fun popularMangaRequest(page: Int) = GET("$baseUrl/hots/page/$page/", headers)
 
     override fun popularMangaParse(response: Response): MangasPage {
         val document = response.asJsoup().also(::parseGenres)
-        val mangas = document.select("article.wp-manga").map { element ->
+        val mangas = document.select(".cardlist .pb-2 a").map { element ->
             SManga.create().apply {
-                val link = element.selectFirst(Evaluator.Tag("h2"))!!.child(0)
-                url = getKey(link.attr("href"))
-                title = link.ownText()
-                thumbnail_url = element.selectFirst(Evaluator.Tag("img"))!!.imgSrc
+                title = element.selectFirst("h3")!!.ownText()
+                thumbnail_url = element.selectFirst("img")?.imgSrc
+                setUrlWithoutDomain(element.absUrl("href"))
```

Comment:
would old urls still work? I belive `getKey` extracts specific part of the url to save to db

Reply:
Old urls are not reached. Example
Current url:
```sh
https://m.baozimh.one/manga/woduzishengji-duburedicestudio-a04
```
Url from getKey (not working)
```
https://m.baozimh.one/manga/woduzishengji-duburedicestudio
```